integreat_cms/cms/utils/zammad.py

```diff
@@ -0,0 +1,243 @@
+        :return: User e-mail address
+        """
+        return self.zammad_request("GET", "/api/v1/users/me").json()["login"]
+
+    def get_zammad_ticket_messages(self) -> list[dict]:
+        """
+        Get Zammad ticket articles
+
+        :return: list of Zammad articles (chat messages)
+        """
+        return self.zammad_request(
+            "GET",
+            f"/api/v1/ticket_articles/by_ticket/{self.zammad_id}",
+        ).json()
+
+    @property
+    def messages(self) -> list[dict]:
+        """
+        Return all messages stored in Zammad for this ticket
+
+        :return: formatted chat messages
+        """
+        if (response := cache.get(f"{self.region.slug}_{self.device_id}")) is None:
+            response = self.get_zammad_ticket_messages()
+        cache.set(f"{self.region.slug}_{self.device_id}", response, 3600)
+        keys_to_keep = [
+            "status",
+            "error",
+            "id",
+            "body",
+            "user_is_author",
+            "automatic_answer",
+            "role",
+            "content",
+            "created_at",
+        ]
+        messages = []
+        for message in response:
+            message["role"] = "user" if message["sender"] == "Customer" else "agent"
+            message["automatic_answer"] = (
+                message["subject"] == "automatically generated message"
+            )
+            message["user_is_author"] = message["role"] == "user"
+            message["content"] = message["body"]
+            reduced_message = {
+                key: message[key] for key in message if key in keys_to_keep
+            }
+            messages.append(reduced_message)
+        return messages
+
+    def save_message(
+        self, message: str, internal: bool, automatic_message: bool
+    ) -> bool:
+        """
+        Add a new message
+
+        :param message: message text to be saved
+        :param internal: true if message shoud not be visible to app user
+        :param automatic_message: true if message does not originate from a human
+        :return: success
+        """
+        cache.delete(f"{self.region.slug}_{self.device_id}")
+        return (
+            self.zammad_request(
+                "POST",
+                "/api/v1/ticket_articles",
+                {
+                    "ticket_id": self.zammad_id,
+                    "body": message,
+                    "internal": internal,
+                    "automatic_message": automatic_message,
+                    "subject": (
+                        "app user message"
+                        if not automatic_message
+                        else "automatically generated message"
+                    ),
+                    "content_type": "text/html",
+                    "type": "web",
+                    "sender": "Customer" if not automatic_message else "Agent",
+                },
+            ).status_code
+            == 200
```

Comment:
Since `zammad_request` can raise errors, this should be wrapped in `try` before accessing  `status_code`. Alternatively, only log the exception in `zammad_request`, but return the status regardless.

Reply:
https://github.com/digitalfabrik/integreat-cms/pull/3496/commits/8c8f7b4404fb6afe3dee55ceae2311bc094b9495